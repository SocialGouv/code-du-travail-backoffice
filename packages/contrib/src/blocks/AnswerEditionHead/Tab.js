import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import _Icon from "../../elements/Icon";

const Container = styled(Flex)`
  background-color: ${props => (props.isActive ? "white" : "lightgray")};
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
  border-right: solid 1px gray;
  border-top: solid 1px gray;
  cursor: pointer;
  padding: 0.25rem 0.5rem;
`;

const Icon = styled(_Icon)`
  margin-right: 0.5rem;
`;

export default ({ children, icon, isActive, onClick }) => (
  <Container alignItems="center" isActive={isActive} onClick={onClick}>
    <Icon icon={icon} isSmall />
    {children}
  </Container>
);
