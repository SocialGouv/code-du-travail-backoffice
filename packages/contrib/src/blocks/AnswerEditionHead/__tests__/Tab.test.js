import React from "react";
import { fireEvent, render } from "@testing-library/react";

import Tab from "../Tab";

describe("[Contrib] blocks/AnswerEditionHead/<Tab />", () => {
  const props = {
    children: "A Tab Text",
    icon: "book",
    isActive: false,
    onClick: jest.fn()
  };

  const { asFragment, container, getByText } = render(<Tab {...props} />);
  const firstRender = asFragment();

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should trigger onClick() with the expected param", () => {
    fireEvent.click(getByText(props.children));

    expect(props.onClick).toHaveBeenCalled();
  });

  it("should match snapshot diff with `isActive` prop", () => {
    const { asFragment } = render(<Tab {...props} isActive />);

    expect(firstRender).toMatchDiffSnapshot(asFragment());
  });
});
