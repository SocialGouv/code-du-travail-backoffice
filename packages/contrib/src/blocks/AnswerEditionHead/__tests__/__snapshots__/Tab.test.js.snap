// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/AnswerEditionHead/<Tab /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 hyrWDR"
  >
    A Tab Text
  </div>
</div>
`;

exports[`[Contrib] blocks/AnswerEditionHead/<Tab /> should match snapshot diff with \`isActive\` prop 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <div
-     class=\\"sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 hyrWDR\\"
+     class=\\"sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 jSKpAZ\\"
    >
      A Tab Text
    </div>
  </DocumentFragment>"
`;
